src/plugin.ts

```diff
@@ -0,0 +1,44 @@
+import { createRequire } from "node:module";
+
+import { rule as orderProperties } from "./rules/order-properties.js";
+import { rule as preferRepositoryShorthand } from "./rules/prefer-repository-shorthand.js";
+import { rule as sortCollections } from "./rules/sort-collections.js";
+import { rule as uniqueDependencies } from "./rules/unique-dependencies.js";
+import { rule as validLocalDependency } from "./rules/valid-local-dependency.js";
+import { rule as validName } from "./rules/valid-name.js";
+import { rule as validPackageDef } from "./rules/valid-package-def.js";
+import { rule as validRepositoryDirectory } from "./rules/valid-repository-directory.js";
+import { rule as validVersion } from "./rules/valid-version.js";
+
+const require = createRequire(import.meta.url || __filename);
+
+const { name, version } = require("../package.json") as {
+	name: string;
+	version: string;
+};
+
+const allRules = {
```

Comment:
[Naming] Super nitpicky, but: I found it confusing that this is named `allRules` and then elsewhere is always referred to as `rules`. What do you think about renaming to `rules`?

```suggestion
const rules = {
```